File: cloud/Core/Identity/Authorization/Role.cs
// Copyright © Spatial Corporation. All rights reserved.

using Spatial.Persistence;

namespace Spatial.Identity.Authorization;

/// <summary>
/// ...
/// </summary>
[Collection("roles")]
public class Role : Resource
{
    /// <summary>
    /// The role's display name.
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    /// What users with this role do.
    /// </summary>
    public string Description { get; set; }

    /// <summary>
    /// The color associated with the role.
    /// </summary>
    public string? Color { get; set; }
}

File: interface/data/roles/Role.ts
// Copyright © Spatial Corporation. All rights reserved.

import { Resource } from "..";

export type Role = Resource<{
  /**
   * The role's display name.
   */
  name: string;

  /**
   * The color associated with the role.
   */
  color: string;

  /**
   * What users with this role do.
   */
  description?: string;
}>;
